packages/echo/src/client.test.ts

```diff
@@ -166,60 +107,65 @@ describe('Echo Client', () => {
         }));
       });
 
-      const discovery = echo.discover();
+      client.addWorkflows([newWorkflow]);
+
+      // wait for discovery to finish
+      await new Promise((resolve) => setTimeout(resolve, 1));
```

Comment:
@rifont so here is a problem with the non awaited workflow method that needs to be resolved in a timely manner. Because this test defines a pretty complex workflow, not all the async methods have time to finish before the `discover()` method below is called. Waiting even 1 millisecond 'fixes' this. 

I've confirmed that this is the case in next as well with the old way of defining workflows. If I remove the await from the `client.workflow()` calls, I still need this wait time to have this particular test pass.

I've tried to play around with the code but no real resolution came to my mind. In reality, the user can do operations in the workflow definition that take an arbitrary amount of time, and discovery won't pick up depending steps until those promises are fulfilled.

How would we target this?